File: .codegen/python/lookup.py
from typing import Final

#pragma once
{{range .lookup.doors.modes}}
{{constant .mode}}: Final[int] = {{.code}}
{{- end}}
{{range .lookup.events.directions}}
DIRECTION_{{constant .direction}}: Final[int] = {{.code}}
{{- end}}
{{range .lookup.events.types}}
EVENT_TYPE_{{constant .type}}: Final[int] = {{.code}}
{{- end}}
{{range .lookup.events.reasons}}
EVENT_REASON_{{constant .reason}}: Final[int] = {{.code}}
{{- end}}

File: .codegen/dotnet/lookup.cs
using System;
using System.Runtime.InteropServices;
using System.Collections.Generic;
using System.Text;

public class DoorMode {
{{- range .lookup.doors.modes}}
    public const byte {{CamelCase .mode}} = {{.code}};
{{- end}}
}

public class Direction {
{{- range .lookup.events.directions}}
    public const byte {{CamelCase .direction}} = {{.code}};
{{- end}}
}

public class EventType {
{{- range .lookup.events.types}}
    public const byte {{CamelCase .type}} = {{.code}};
{{- end}}
}

public class EventReason {
{{- range .lookup.events.reasons}}
    public const byte {{CamelCase .reason}} = {{.code}};
{{- end}}
}
